### PR TITLE
Literature Submission bugs

### PR DESCRIPTION
**Pull request by harunurhan**

### CODE REVIEW

File: ui/src/submissions/literature/components/ArticleForm.jsx [hunk 1]
```diff
@@ -65,7 +66,7 @@ class ArticleForm extends Component {
             />
             <Field name="volume" label="Volume" component={TextField} />
             <Field name="issue" label="Issue" component={TextField} />
-            <Field name="year" label="Year" component={TextField} />
+            <Field name="year" label="Year" component={NumberField} />
```
Comment (drjova) [inline ui/src/submissions/literature/components/ArticleForm.jsx]: do we have a tests for these value types?

File: ui/src/submissions/literature/components/LiteratureSubmission.jsx [hunk 2]
```diff
@@ -37,60 +43,75 @@ const FORMS_BY_DOC_TYPE = {
...
-    this.onFormikSubmit = this.onFormikSubmit.bind(this);
-  }
+  const [isInitilValid, setInitialValid] = useState(false);
+  useEffect(
+    () => {
+      const validate = async () => {
+        setInitialValid(await schema.isValid(initialValues));
+      };
+      validate();
+    },
```
Comment (drjova) [inline ui/src/submissions/literature/components/LiteratureSubmission.jsx]: mann it's too complicated to read :(

Comment (harunurhan) [reply ui/src/submissions/literature/components/LiteratureSubmission.jsx]: yeah, I agree, `async` calls in effect hook is pretty weird. I'll create a custom hook for this, to simplify here.

File: ui/src/submissions/literature/components/ArticleForm.jsx (hunk 1 above)
Comment (harunurhan) [reply ui/src/submissions/literature/components/ArticleForm.jsx]: tested via `e2e`, test is probably not filling the year field.

Comment (harunurhan) [reply ui/src/submissions/literature/components/ArticleForm.jsx]: it's visible in screenshot though, that year is a number field.

Comment (drjova) [reply ui/src/submissions/literature/components/ArticleForm.jsx]: can we add the year field?

**Comment by harunurhan** [comment]
@drjova addressed the comments

**Pull request merged**